python/pythonlab/neighborhood/neighborhood/support/neighborhood_signal_message.py

```diff
@@ -0,0 +1,51 @@
+import json
+from .signal_message_type import SignalMessageType
+
+class NeighborhoodSignalMessage:
+    """
+    All Neighborhood signal messages will use the following format:
+        {
+            "type": <SignalMessageType>,
+            "value": <"message data">,
+            "detail": <dict>
+        }
+    """
+
+    def __init__(self, type: SignalMessageType, value: str, detail: dict = None):
+        self.type = type
+        self.value = value
+        self.detail = detail
+
+    def get_type(self) -> SignalMessageType:
```

Comment:
I'm not sure if we need these getters. All fields in python are public so you can call `message.type` already.